mlflow/gateway/schemas/chat.py

```diff
@@ -1,17 +1,119 @@
-from typing import Literal, Optional
+from typing import Literal, Optional, Union
 
-from pydantic import Field
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
 
 from mlflow.gateway.base_models import RequestModel, ResponseModel
-from mlflow.gateway.config import IS_PYDANTIC_V2
+from mlflow.utils import IS_PYDANTIC_V2
 
 
-class RequestMessage(RequestModel):
+class TextContentPart(RequestModel):
+    type: Literal["text"]
+    text: str
+
+
+class ImageUrl(RequestModel):
```

Comment:
Is there any reason we use `RequestModel` rather than the `BaseModel`? 

The docstring of the `BaseModel` says
```
    # Allow extra fields for pydantic request models, e.g. to support
    # vendor-specific embeddings parameters
```

I don't think we need to support extra fields here? If possible, less number of abstraction is better.

Reply:
ah perfect, let me switch